pkg/converters/ingress/ingress.go

```diff
@@ -952,7 +952,8 @@ func (c *converter) syncBackendEndpointCookies(backend *hatypes.Backend) {
 }
 
 func (c *converter) syncBackendEndpointHashes(backend *hatypes.Backend) {
-	if !backend.Server.AssignID {
+	mapper := c.backendAnnotations[backend]
+	if mapper != nil && !mapper.Get(ingtypes.BackAssignBackendServerID).Bool() {
```

Comment:
Actually shouldn't this be `if mapper == nil || !mapper.Get(ingtypes.BackAssignBackendServerID).Bool()` ?

Reply:
@jcmoraisjr

Reply:
ouch, yes, makes sense. in fact this `mapper != nil` is a "just in case", it should never return nil for a known backend, but your proposal is in fact the correct way to implement this.

Reply:
All set.